src/atomate2/vasp/jobs/phonons.py

```diff
@@ -0,0 +1,418 @@
+    full_born: bool = True,
+    npoints_band: int = 100,
+    kpoint_density_dos: int = 7000,
+    tol_imaginary_modes: float = 1e-5,
+    tmin=0,
+    tmax=500,
+    tstep=10,
+    units="THz",
+    img_format="eps",
+):
+    """
+    Compute phonon band structures and density of states.
+
+    Parameters
+    ----------
+
+    """
+    # TODO: move his part to another class?
+    def get_kpath(structure: Structure, kpath_scheme: str, **kpath_kwargs):
+        """
+        get high-symmetry points in k-space
+        Args:
+            structure: Structure Object
+        Returns:
+        """
+
+        if kpath_scheme in [
+            "setyawan_curtarolo",
+            "hinuma",
+            "latimer_munro",
+            "all_pymatgen",
+        ]:
+            if kpath_scheme == "all_pymatgen":
+                kpath_scheme = "all"
+            highsymmkpath = HighSymmKpath(
+                structure, path_type=kpath_scheme, **kpath_kwargs
+            )
+            kpath = highsymmkpath.kpath
+        elif kpath_scheme == "seekpath":
+            highsymmkpath = KPathSeek(structure, **kpath_kwargs)
+            kpath = highsymmkpath._kpath
+
+        path = copy.deepcopy(kpath["path"])
+
+        for ilabelset, labelset in enumerate(kpath["path"]):
+            for ilabel, label in enumerate(labelset):
+                path[ilabelset][ilabel] = kpath["kpoints"][label]
+        return kpath["kpoints"], path
+
+    # have to regenerate this object as I cannot make it a job output
+    # TODO: other way?
+    phonon = get_phonon_object(
+        structure=structure,
+        supercell_matrix=supercell_matrix,
+        displacement=displacement,
+        sym_reduce=sym_reduce,
+        symprec=symprec,
+        use_standard_primitive=use_standard_primitive,
+        kpath_scheme=kpath_scheme,
+        code=code,
+    )
+
+    set_of_forces = [np.array(forces) for forces in displacement_data["forces"]]
+
+    if born is not None:
+        if full_born:
+            # TODO: if this is a good way when user provide data
+            borns, epsilon, atom_indices = elaborate_borns_and_epsilon(
+                ucell=get_phonopy_structure(structure),
+                borns=np.array(born),
+                epsilon=np.array(epsilon_static),
+                symprec=symprec,
+                primitive_matrix=phonon.primitive_matrix,
+                supercell_matrix=phonon.supercell_matrix,
+            )
+        else:
+            borns = born
+        if code == "vasp":
+            phonon.nac_params = {
+                "born": borns,
+                "dielectric": epsilon,
+                "factor": 14.399652,
+            }
+    phonon.produce_force_constants(forces=set_of_forces)
+
+    # Currently, the setting of the born
+    # charges does not work without this! I am not sure why!
+    # I don't see it
+    # I think I am missing something in the code
+    phonon.save("phonopy.yaml")
+    phonon = load("phonopy.yaml")
```

Comment:
@utf : I currently have a problem here. I can only get correct bandstructures with NAC correction when I save the phonon object and then reload. There should be another way but I currently don't find the bug. If you have any idea, I am happy to hear.

Reply:
I am attaching minimal example where it does not happen. I can currently not reproduce this bug outside of the workflow. 
[minimal_phonopy.zip](https://github.com/materialsproject/atomate2/files/8879780/minimal_phonopy.zip)

Reply:
I'm afraid I can't see what is going wrong!

Can you confirm that you're using the same phonopy version outside of the workflow?

Reply:
I will confirm this and check again but thanks already for having a look!

Reply:
I have upgraded to the latest phonopy version. It is still happening. I think I will continue with this file creation part for now and try to finalize the document part first. 
![phonon_band_structure-01](https://user-images.githubusercontent.com/22094846/173116451-18e758a0-3276-4a6c-900e-72eafa7f83de.png)

Reply:
This is the script that I am using for the workflow. I might need to try it in a completely new environment to make sure this is also not the reason for this problem.
[start_workflow.zip](https://github.com/materialsproject/atomate2/files/8880814/start_workflow.zip)

Reply:
It seems like the workflow is using the same symprec throughout but it is quite high. I wonder if it will fix things if you use 1e-5?

Reply:
I will try! Thanks!

Reply:
You were correct. It was the symprec... I have now introduced a second symprec setting. 

I will try to further clean up the code in the next days and check the kpath setting.